src/robusta/core/sinks/pagerduty/pagerduty_sink.py

```diff
@@ -0,0 +1,119 @@
+import requests
+from robusta.core.reporting.blocks import (
+    HeaderBlock,
+    ListBlock,
+    JsonBlock,
+    KubernetesDiffBlock,
+    MarkdownBlock,
+    TableBlock,
+)
+from .pagerduty_sink_params import PagerdutyConfigWrapper
+from ...reporting.base import Finding, BaseBlock, FindingSeverity
+
+
+from ..sink_base import SinkBase
+
+
+class PagerdutySink(SinkBase):
+    def __init__(self, sink_config: PagerdutyConfigWrapper, registry):
+        super().__init__(sink_config.pagerduty_sink, registry)
+        self.url = sink_config.pagerduty_sink.url        
+        self.api_key = sink_config.pagerduty_sink.api_key        
+    
+    def __to_pagerduty_severity_type(severity: FindingSeverity):
+        # must be one of [critical, error, warning, info]
+        # Default Incident Urgency is interpreted as [HIGH, HIGH, LOW, LOW]
+        # https://support.pagerduty.com/docs/dynamic-notifications
+        if severity == FindingSeverity.HIGH:
+            return "critical"
+        elif severity == FindingSeverity.MEDIUM:
+            return "error"        
+        elif severity == FindingSeverity.LOW:
+            return "warning"
+        elif severity == FindingSeverity.INFO:
+            return "info"        
+        elif severity == FindingSeverity.DEBUG:
+            return "info"      
+        else:
+            return "critical"
+        
+    def __to_pagerduty_status_type(title: str):
+        # very dirty implementation, I am deeply sorry
+        # must be one of [trigger, acknowledge or resolve]
+        if title.startswith('[RESOLVED]'):
+            return "resolve"
+        else:
+            return "trigger"
+
+    def write_finding(self, finding: Finding, platform_enabled: bool):
+        custom_details: dict = {}
+
+        if platform_enabled:
+            custom_details["🔎 Investigate"] = finding.investigate_uri
+
+            if finding.add_silence_url:
+                custom_details[
+                    "🔕 Silence"
+                ] = finding.get_prometheus_silence_url(self.cluster_name)
+
+        # custom fields
```

Comment:
This looks good, I just wanted to verify that I understood it correctly. The fields in `custom_details` are all fields that you defined, which don't have an inherent meaning in PagerDuty itself, right?

If so, maybe add a comment explaining that. Looks good though.

Reply:
Yes, I was inspired by another sink module (victorops or datadog etc) ;)

Reply:
Thanks